### PR TITLE
Chat

### PR DESCRIPTION
**Pull request by wood-archer**
Knockdown list
---
- [x] add username to credentials
- [x] remove basic chat example to not conflict with the definitive one
- [x] implement database design
- [x] add the ability to create conversations
- [ ] fix broken tests

### CODE REVIEW

File: lib/turing/chat.ex [hunk 1]
```diff
@@ -6,24 +6,210 @@ defmodule Turing.Chat do
...
+
+      iex> change_conversation(conversation)
+      %Ecto.Changeset{source: %Conversation{}}
+
+  """
+  def change_conversation(%Conversation{} = conversation) do
+    Conversation.changeset(conversation, %{})
+  end
+
+  alias Turing.Chat.ConversationMember
```
Comment (justuseapen) [inline lib/turing/chat.ex]: Does this alias belong here? Should we keep aliases at the top of the file?

File: lib/turing/chat.ex [hunk 2]
```diff
@@ -108,16 +290,295 @@ defmodule Turing.Chat do
...
+
+      iex> change_emoji(emoji)
+      %Ecto.Changeset{source: %Emoji{}}
+
+  """
+  def change_emoji(%Emoji{} = emoji) do
+    Emoji.changeset(emoji, %{})
+  end
+
+  alias Turing.Chat.MessageReaction
```
Comment (justuseapen) [inline lib/turing/chat.ex]: See question above.

File: lib/turing/chat/conversation_member.ex [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  use Ecto.Schema
+  import Ecto.Changeset
+
+  alias Turing.Accounts.User
+  alias Turing.Chat.Conversation
+
+  @primary_key {:id, :binary_id, autogenerate: true}
+  @foreign_key_type :binary_id
+  schema "chat_conversation_members" do
+    field :owner, :boolean, default: false
```
Comment (justuseapen) [inline lib/turing/chat/conversation_member.ex]: What is this field for?

File: lib/turing/chat/emoji.ex [hunk 4]
```diff
@@ -0,0 +1,20 @@
+defmodule Turing.Chat.Emoji do
```
Comment (justuseapen) [inline lib/turing/chat/emoji.ex]: 😂 necessary

File: lib/turing/chat/message.ex [hunk 5]
```diff
@@ -2,19 +2,27 @@ defmodule Turing.Chat.Message do
   use Ecto.Schema
   import Ecto.Changeset
 
+  alias Turing.Accounts.User
+  alias Turing.Chat.{Conversation, SeenMessage, MessageReaction}
+
   @primary_key {:id, :binary_id, autogenerate: true}
   @foreign_key_type :binary_id
-  schema "messages" do
+  schema "chat_messages" do
```
Comment (justuseapen) [inline lib/turing/chat/message.ex]: why not just "messages"?

File: lib/turing/chat.ex (hunk 1 above)
Comment (TheCraftedGem) [reply lib/turing/chat.ex]: It'd make sense to keep it at the top.

File: lib/turing/chat/message.ex (hunk 5 above)
Comment (wood-archer) [reply lib/turing/chat/message.ex]: To keep context tracking on a db level mostly. Not a big deal though.

File: lib/turing/chat/conversation_member.ex (hunk 3 above)
Comment (wood-archer) [reply lib/turing/chat/conversation_member.ex]: For each conversation, one user can be its owner, who'll be able to e.g. close it.

File: lib/turing/chat.ex (hunk 1 above)
Comment (wood-archer) [reply lib/turing/chat.ex]: Yeah. I was coding late night so I might have missed that :sweat_smile:

**Pull request merged**